iaso/api/workflows.py

```diff
@@ -0,0 +1,361 @@
+        }
+
+    for v in entity_type.workflow.workflow_versions.all():
+        if (
+            v.follow_ups.count() > 0
+            and not v.follow_ups.filter(forms__projects__account=user.iaso_profile.account).exists()
+        ):
+            return {
+                "can_access": False,
+                "reason": f"User {user.username} Cannot Access FollowUps Form for Workflow Version {v.pk}",
+            }
+
+        if v.changes.count() > 0 and not v.changes.filter(form__projects__account=user.iaso_profile.account).exists():
+            return {
+                "can_access": False,
+                "reason": f"User {user.username} Cannot Access Changes Form for Workflow Version {v.pk}",
+            }
+
+    return {"can_access": True}
+
+
+def user_can_access_all_forms_of_workflow_version(workflow_version: WorkflowVersion, user: User):
+    if (
+        workflow_version.workflow.entity_type.reference_form is not None
+        and not workflow_version.workflow.entity_type.reference_form.projects.filter(
+            account=user.iaso_profile.account
+        ).exists()
+    ):
+
+        return {
+            "can_access": False,
+            "reason": f"User {user.username} Cannot Access Reference Form for Workflow Version {WorkflowVersion.pk}",
+        }
+
+    if (
+        workflow_version.follow_ups.count() > 0
+        and not workflow_version.follow_ups.filter(forms__projects__account=user.iaso_profile.account).exists()
+    ):
+        return {
+            "can_access": False,
+            "reason": f"User {user.username} Cannot Access FollowUps Form for Workflow Version {workflow_version.pk}",
+        }
+
+    if (
+        workflow_version.changes.count() > 0
+        and not workflow_version.changes.filter(form__projects__account=user.iaso_profile.account).exists()
+    ):
+        return {
+            "can_access": False,
+            "reason": f"User {user.username} Cannot Access Changes Form for Workflow Version {workflow_version.pk}",
+        }
+
+    return {"can_access": True}
+
+
+class WorkflowVersionViewSet(GenericViewSet):
+    """Workflow API
+    POST /api/workflow/{entity_type_id}/new/?version_id=XXX
+    if version_id is provided, copies the content of the version referred to by the version_id
+    if version_id is NOT provided, creates a new workflow from scratch (empty).
+    The new version is always in DRAFT
+
+    GET /api/workflow/{entity_type_id}/?version_id=XXX
+    if version_id is provided, it will return details on this specific version.
+    if version_id is NOT provided, it will return the (paginated) list of all versions for this workflow.
+    """
+
+    lookup_field = "entity_type_id"
+    permission_classes = [permissions.IsAuthenticated, HasPermission("menupermissions.iaso_workflows")]  # type: ignore
+    serializer_class = WorkflowVersionSerializer
+    pagination_class = WorkflowPaginator
+
+    # filter_backends = [filters.OrderingFilter, DjangoFilterBackend]
```

Comment:
an hesitation or dead code ?

Reply:
Dead code it's gonna be removed ... Anyway I'm doing a big refactoring after a deep review with Olivier and to follow more the URLs and patterns of django DRF